clang/lib/Sema/SemaTemplateDeductionGuide.cpp

```diff
@@ -1398,5 +1807,30 @@ void Sema::DeclareImplicitDeductionGuides(TemplateDecl *Template,
           ->getTemplatedDecl())
       ->setDeductionCandidateKind(DeductionCandidate::Copy);
 
+  CXXRecordDecl *TemplatedDecl = Pattern->getTemplatedDecl();
+  if (getLangOpts().CPlusPlus23 && TemplatedDecl->hasDefinition()) {
```

Comment:
Is that something we should support in C++20 as an extension?

Reply:
Given the uncertainty with open CWG issues and the potential breakage of code on earlier standards versions, I think it's less risky to keep it constrained to C++23 for now. It's probably worth revisiting in the future, though.